src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -86,12 +88,14 @@ internal class LocalGrainDirectory :
         internal readonly CounterStatistic UnregistrationsManyRemoteSent;
         internal readonly CounterStatistic UnregistrationsManyRemoteReceived;
 
-        
         public LocalGrainDirectory(Silo silo)
         {
             log = LogManager.GetLogger("Orleans.GrainDirectory.LocalGrainDirectory");
 
             MyAddress = silo.LocalMessageCenter.MyAddress;
+            isSiloNextInTheRing = (siloAddr, hash, excludeMySelf) => (siloAddr.GetConsistentHashCode() <= hash) &&
+                                                                     (!excludeMySelf || !siloAddr.Equals(MyAddress));
```

Comment:
![image](https://cloud.githubusercontent.com/assets/203839/19461077/1ab014bc-952d-11e6-9d83-a7d15e6f3016.png)

You're right, @jdom. It's a good idea to remove the allocation here, but unless there's something wrong with `membershipRingList.FindLast`, that call can be retained.

Instead, I would suggest creating a `private readonly Predicate<SiloAddress>` field which captures `this` once and passing that to the call here. That way you have no per-call allocations.